plugins/providers/metabase/provider_test.go

```diff
@@ -633,4 +885,901 @@ func TestGrantAccess(t *testing.T) {
+							},
+						},
+					},
+				},
+				URN: providerURN,
+			}
+			a := &domain.Appeal{
+				Resource: &domain.Resource{
+					Type: metabase.ResourceTypeTable,
+					URN:  "table:999.1000",
+					Name: "test-database",
+				},
+				Role:       "test-role",
+				AccountID:  expectedUser,
+				ResourceID: "999",
+				ID:         "999",
+			}
+
+			actualError := p.GrantAccess(pc, a)
+
+			assert.EqualError(t, actualError, expectedError.Error())
+		})
+
+		t.Run("should return nil error if granting table access is successful", func(t *testing.T) {
+			providerURN := "test-provider-urn"
+			crypto := new(mocks.Crypto)
+			client := new(mocks.MetabaseClient)
+			expectedTable := &metabase.Table{
+				Name: "test-table",
+				ID:   1000,
+				DbId: 999,
+			}
+			expectedUser := "test@email.com"
+			expectedRole := "viewer"
+			logger := log.NewLogrus(log.LogrusWithLevel("info"))
+			p := metabase.NewProvider("", crypto, logger)
+
+			p.Clients = map[string]metabase.MetabaseClient{
+				providerURN: client,
+			}
+			client.On("GrantTableAccess", expectedTable, expectedUser, expectedRole, mock.Anything).Return(nil).Once()
+
+			d := []*metabase.GroupResource{{Urn: "database:1", Permissions: []string{"read", "write"}}}
+			c := []*metabase.GroupResource{{Urn: "collection:1", Permissions: []string{"read", "write"}}}
+			group := metabase.Group{Name: "All Users", DatabaseResources: d, CollectionResources: c}
+			client.On("GetGroups").Return([]*metabase.Group{&group},
+				metabase.ResourceGroupDetails{"database:1": {{"urn": "group:1", "permissions": []string{"read", "write"}}}},
+				metabase.ResourceGroupDetails{"collection:1": {{"urn": "group:1", "permissions": []string{"write"}}}}, nil).Once()
+
+			pc := &domain.ProviderConfig{
+				Credentials: metabase.Credentials{
+					Host:     "localhost",
+					Username: "test-username",
+					Password: "test-password",
+				},
+				Resources: []*domain.ResourceConfig{
+					{
+						Type: metabase.ResourceTypeTable,
+						Roles: []*domain.Role{
+							{
+								ID:          "viewer",
+								Permissions: []interface{}{expectedRole},
+							},
+						},
+					},
+				},
+				URN: providerURN,
+			}
+			a := &domain.Appeal{
+				Resource: &domain.Resource{
+					Type: metabase.ResourceTypeTable,
+					URN:  "table:999.1000",
+					Name: "test-table",
+				},
+				Role:       "viewer",
+				AccountID:  expectedUser,
+				ResourceID: "999",
+				ID:         "999",
+			}
+
+			actualError := p.GrantAccess(pc, a)
+
+			assert.Nil(t, actualError)
+		})
+
+	})
+}
+
+func TestRevokeAccess(t *testing.T) {
+	t.Run("should return an error if there is an error in getting permissions", func(t *testing.T) {
+		var permission metabase.Permission
+		invalidPermissionConfig := map[string]interface{}{}
+		invalidPermissionConfigError := mapstructure.Decode(invalidPermissionConfig, &permission)
+
+		testcases := []struct {
+			resourceConfigs []*domain.ResourceConfig
```

Comment:
```suggestion
		testcases := []struct {
		    name            string
			resourceConfigs []*domain.ResourceConfig
```
add name for test case description